### PR TITLE
Add unit tests for teacher application secion 4: workshops selection

### PR DESCRIPTION
**Pull request by aoby**
There's a bit of complex logic in section 4 of the teacher application: loading assigned workshops via API based on school & course, then sometimes additionally loading alternate workshops. This adds unit tests for that logic.

### CODE REVIEW

**Comment by aoby** [comment]
[Tests for teacher application](https://trello.com/c/uiFmdrgE/267-tests-for-teacher-application)

File: apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx [hunk 1]
```diff
@@ -183,8 +184,8 @@ export default class Section4SummerWorkshop extends ApplicationFormComponent {
         return (
           <div>
             <h5>
-              Your region’s assigned summer workshop will be
-              {' ' + this.state.partnerWorkshops[0].dates + ' '} in
+                Your region’s assigned summer workshop will be
```
Comment (mehalshah) [inline apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx]: Intentional? Are you trying to indent this paragraph?

Comment (mehalshah) [inline apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx]: Prefer using string interpolation for this

Comment (aoby) [reply apps/src/code-studio/pd/application/teacher1819/Section4SummerWorkshop.jsx]: unintentional. will fix

**Pull request merged**